### PR TITLE
HTTP proxy sending 504 status code

### PR DESCRIPTION
**Pull request by jitsni**

### CODE REVIEW

**Comment by jitsni** [comment]
@msalavastru good work. I am creating a new PR, see if you have any comments

**Comment by jitsni** [comment]
One test that is missing: when server takes too much time to respond, then proxy need to send 504 to client. May be it can be addressed when we do DOS story.

File: specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+accepted
+connected
+
+read method "GET"
+read header "Host" "localhost:8080"
+read notify GOT_REQUEST
+read closed
+
+write await GOT_RESPONSE
+write status "504" "OK"
```
Comment (msalavastru) [inline specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt]: This should be **write status "504" "Gateway Timeout"**

File: specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#
+# client <--> proxy
+#
+accept http://localhost:8080/index.html
+accepted
+connected
+
+read method "GET"
```
Comment (msalavastru) [inline specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt]: Should add **read version "HTTP/1.1"**

File: specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt (hunk 1 above)
Comment (jitsni) [inline specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt]: Created a PR for this.

File: specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt (hunk 2 above)
Comment (jitsni) [inline specification/http/src/main/scripts/org/kaazing/specification/http/rfc7231/server.error/proxy.should.return.504.response.when.server.disconnects/proxy.rpt]: Not that important to check version in this test, but included it in the PR

**Pull request merged**